File: ulohy/21-brigady/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let m;
const hoursWorked = [];
let hourlyRatesLoaded = 0;
let salary = 0;

rl.on("line", (line) => {
    if (m === undefined) {
        m = parseInt(line);
        return;
    }

    if (hoursWorked.length < m) {
        if (line === "") {
            return;
        }

        hoursWorked.push(parseInt(line));
        return;
    }

    if (hourlyRatesLoaded < m) {
        if (line === "") {
            return;
        }

        const hourlyRate = parseInt(line);
        salary += hoursWorked[hourlyRatesLoaded] * hourlyRate;
        hourlyRatesLoaded++;
    }

    if (hourlyRatesLoaded >= m) {
        console.log(salary);
        rl.close();
        process.exit(0);
    }
});

File: ulohy/21-brigady/reseni/main.cs
using System;

internal class Program
{
    static void Main(string[] args)
    {
        int m = ReadInt();
        Console.ReadLine();

        int[] hoursWorked = new int[m];
        for (int i = 0; i < m; i++)
        {
            hoursWorked[i] = ReadInt();
        }

        Console.ReadLine();

        int salary = 0;
        for (int i = 0; i < m; i++)
        {
            int hourlyRate = ReadInt();
            salary += hoursWorked[i] * hourlyRate;
        }

        Console.WriteLine(salary);
    }

    static int ReadInt()
    {
        return int.Parse(Console.ReadLine());
    }
}
